generated/nidaqmx/_base_interpreter.py

```diff
@@ -0,0 +1,1621 @@
+            voltage_excit_val, custom_scale_name):
+        pass
+
+    def create_tedsairtd_chan(
+            self, task, physical_channel, name_to_assign_to_channel, min_val,
+            max_val, units, resistance_config, current_excit_source,
+            current_excit_val):
+        pass
+
+    @staticmethod
+    def create_watchdog_timer_task(
+            self, device_name, session_name, task, timeout, lines, exp_state,
+            initialization_behavior, new_session_initialized):
+        pass
+
+    def create_watchdog_timer_task_ex(
+            self, device_name, session_name, task, timeout,
+            initialization_behavior, new_session_initialized):
+        pass
+
+    def delete_network_device(self, device_name):
+        pass
+
+    def delete_saved_global_chan(self, channel_name):
+        pass
+
+    def delete_saved_scale(self, scale_name):
+        pass
+
+    def delete_saved_task(self, task_name):
+        pass
+
+    def device_supports_cal(self, device_name, cal_supported):
+        pass
+
+    def disable_ref_trig(self, task):
+        pass
+
+    def disable_start_trig(self, task):
+        pass
+
+    def disconnect_terms(self, source_terminal, destination_terminal):
+        pass
+
+    def export_signal(self, task, signal_id, output_terminal):
+        pass
+
+    def get_ai_chan_cal_cal_date(
+            self, task, channel_name, year, month, day, hour, minute):
+        pass
+
+    def get_ai_chan_cal_exp_date(
+            self, task, channel_name, year, month, day, hour, minute):
+        pass
+
+    def get_analog_power_up_states(
+            self, device_name, channel_name, state, channel_type):
+        pass
+
+    def get_analog_power_up_states_with_output_type(
+            self, channel_names, state_array, channel_type_array, array_size):
+        pass
+
+    def get_arm_start_trig_timestamp_val(self, task, data):
+        pass
+
+    def get_arm_start_trig_trig_when(self, task, data):
+        pass
+
+    def get_auto_configured_cdaq_sync_connections(
+            self, port_list, port_list_size):
+        pass
+
+    def get_buffer_attribute_uint32(self, task, attribute, value):
+        pass
+
+    def get_cal_info_attribute_bool(self, device_name, attribute, value, size):
+        pass
+
+    def get_cal_info_attribute_double(
+            self, device_name, attribute, value, size):
+        pass
+
+    def get_cal_info_attribute_string(
+            self, device_name, attribute, value, size):
+        pass
+
+    def get_cal_info_attribute_uint32(
+            self, device_name, attribute, value, size):
+        pass
+
+    def get_chan_attribute_bool(self, task, channel, attribute, value, size):
+        pass
+
+    def get_chan_attribute_double(self, task, channel, attribute, value, size):
+        pass
+
+    def get_chan_attribute_double_array(
+            self, task, channel, attribute, value, size):
+        pass
```

Comment:
Python lists and numpy arrays are mutable, so I don't think it's critical to represent array out-parameters as return values. However, we should remove the `size` parameter. The library interpreter should do the "double call" or "IVI dance".

Reply:
Ok. But some of the functions has multiple out parameters for a single function. which include multiple arrays as return values.
- `GetAnalogPowerUpStatesWithOutputType`
- `ReadCtrTicks`

In that case, should we need to form a `Tuple(List[int], List[float], ...)` as return value for the functions?
However, I have removed the size parameter as suggested.

Reply:
> In that case, should we need to form a Tuple(List[int], List[float], ...) as return value for the functions?

If returning, yes. Conveniently, this still turns into
```python
    return foo, bar, # ...
```

If *mutating the passed list(s)*, the input parameter should ask for a `List`, AND the docstring should note that it will add to the parameter object.

Reply:
Sure. But should we need to type annotate the return values in `BaseInterpreter` like this,

```
def get_analog_power_up_states_with_output_type(
            self, channel_names, array_size): -> Tuple(List[Float], List[int], ...):
    raise NotImplementedError
``

Reply:
If *returning* the lists, yes. they should be documents as a `) -> Tuple(` return.

The exact value of that I would need to know more about the intended api. If it's only a list of floats, and a list of ints, then `) -> Tuple(List[float], List[int]):`.

Reply:
Current status:
- This codebase has no type annotations at all.
- I [tried running mypy yesterday](https://github.com/ni/nidaqmx-python/tree/users/bkeryan/mypy/), and `check_untyped_defs` produces a lot of errors. 
- Type annotations are not part of the acceptance criteria for the gRPC feature.

So I don't think we should require type annotations. 

If we can *easily* generate type annotations, they are nice to have. However, they would have more value for public interfaces than for internal stuff like the interpreter.

Maybe I will try generating `.pyi` files at some point.

Reply:
Sure. I have removed the function `get_return_value_for_func()` which was previously used to set return value (type annotation) as `str` for functions.
As suggested, type annotations for individual parameters along with the return value for functions will not be handled as part of this feature.